.github/workflows/main.yml

```diff
@@ -0,0 +1,54 @@
+name: Main
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+jobs:
+  check-workflows:
+    name: Check workflows
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Download actionlint
+        id: download-actionlint
+        run: bash <(curl https://raw.githubusercontent.com/rhysd/actionlint/7fdc9630cc360ea1a469eed64ac6d78caeda1234/scripts/download-actionlint.bash) 1.6.22
+        shell: bash
+      - name: Check workflow files
+        run: ${{ steps.download-actionlint.outputs.executable }} -color
+        shell: bash
+
+  build-lint-test:
+    name: Build, lint, and test
```

Comment:
What do you think about running this job after running `check-workflows`, in case that fails, so we can see the failure sooner? It shouldn't take that long to run that.

Reply:
Wouldn't that result in us seeing errors more slowly? Not sure I follow

Reply:
I mean, I guess it depends on which errors we are looking for. `check-workflows` looks for specific things that may not be visible when running the workflows (if I am understanding the value of this tool correctly). So `build-lint-test` may fail, but in a different way than `check-workflows`. I guess there's no harm in that. Okay, no problem then.

Reply:
What I mean is that if neither depends on the other, they both run without delay and produce output. But if we wait for actionlint to run before the other checks, then the other checks will be delayed or may never run at all in the case of a workflow error.